pyaerocom/io/cachehandler_ungridded.py

```diff
@@ -247,46 +296,59 @@ def delete_all_cache_files(self):
+
+        Returns
+        -------
+        str
+            output file path
         """
         meta = self.cache_meta_info()
 
         if not isinstance(data, UngriddedData):
             raise TypeError('Invalid input, need instance of UngriddedData, '
                             'got {}'.format(type(data)))
-        if len(data.contains_datasets) > 1:
-            raise CacheWriteError('Input UngriddedData object contains '
-                                  'datasets: {}. Can only write single '
-                                  'dataset objects'
-                                  .format(data.contains_datasets))
-        if var_name is None:
+
+        if not var_or_file_name.endswith('.pkl'):
+            var_name = var_or_file_name
+            if len(data.contains_datasets) > 1:
+                raise CacheWriteError('Input UngriddedData object contains '
+                                      'datasets: {}. Can only write single '
+                                      'dataset objects'
+                                      .format(data.contains_datasets))
+            if var_name is None:
+                if len(data.contains_vars) > 1:
+                    raise CacheWriteError('Input UngriddedData object for {} contains '
+                                          'more than one variable: {}. Please '
+                                          'specify which variable should be '
+                                          'cached'
+                                          .format(self.reader.data_id,
+                                                  data.contains_vars))
+                var_name = data.contains_vars[0]
+
+            elif not var_name in data.contains_vars:
+                raise CacheWriteError('Cannot write cache file: variable {} does '
+                                      'not exist in input UngriddedData object'
+                                      .format(var_name))
+
             if len(data.contains_vars) > 1:
-                raise CacheWriteError('Input UngriddedData object for {} contains '
-                                      'more than one variable: {}. Please '
-                                      'specify which variable should be '
-                                      'cached'
-                                      .format(self.reader.data_id,
-                                              data.contains_vars))
-            var_name = data.contains_vars[0]
-
-        elif not var_name in data.contains_vars:
-            raise CacheWriteError('Cannot write cache file: variable {} does '
-                                  'not exist in input UngriddedData object'
-                                  .format(var_name))
-
-        if len(data.contains_vars) > 1:
-            data = data.extract_var(var_name)
-
-        fp = self.file_path(var_name)
+                data = data.extract_var(var_name)
+
+        fp = self.file_path(var_or_file_name, cache_dir=cache_dir)
```

Comment:
This part is a bit confusing after introduction of `var_or_file_name`.
Shouldn't the checks for number of datasets, variables and var_name run even if we call the function with a filename?

Reply:
I agree it is a little confusing, but it is only important to check that for the normal caching (that requires access to the database), since the data in the database may have been changed in the meantime. The new case where also a filename can be provided directly is not supposed to check anything and is sort of just a workaround for storing *any* instance of an `UngriddedData`  object. Rethinking the whole thing, probably it would've been easier to not use `CacheHandlerUngridded` at all for this tast, and just `pickle.dump` directly `UngriddedData` objects directly... but let's leave it for now.

Reply:
I see, thanks for the clarification!